mypy/find_sources.py

```diff
@@ -52,60 +55,87 @@ def keyfunc(name: str) -> Tuple[int, str]:
+        f = get_init_file(self.fscache, arg)
+        if mod_prefix and not f:
+            return []
+        seen = set()  # type: Set[str]
+        sources = []
+        if f and not mod_prefix:
+            top_mod = self.crawl_up(f)
+            mod_prefix = top_mod + '.'
+        if mod_prefix:
+            sources.append(BuildSource(f, mod_prefix.rstrip('.'), None))
+        names = self.fscache.listdir(arg)
+        names.sort(key=keyfunc)
+        for name in names:
+            path = os.path.join(arg, name)
+            if self.fscache.isdir(path):
+                sub_sources = self.expand_dir(path, mod_prefix + name + '.')
+                if sub_sources:
+                    seen.add(name)
+                    sources.extend(sub_sources)
+            else:
+                base, suffix = os.path.splitext(name)
+                if base == '__init__':
+                    continue
+                if base not in seen and '.' not in base and suffix in PY_EXTENSIONS:
+                    seen.add(base)
+                    src = BuildSource(path, mod_prefix + base, None)
+                    sources.append(src)
+        return sources
+
+    def crawl_up(self, arg: str) -> str:
+        """Given a .py[i] filename, return module.
+
+        We crawl up the path until we find a directory without
+        __init__.py[i], or until we run out of path components.
+        """
+        dir, mod = os.path.split(arg)
+        mod = strip_py(mod) or mod
+        base = self.crawl_up_dir(dir)
+        if mod == '__init__' or not mod:
+            mod = base
         else:
-            base, suffix = os.path.splitext(name)
-            if base == '__init__':
-                continue
-            if base not in seen and '.' not in base and suffix in PY_EXTENSIONS:
-                seen.add(base)
-                src = BuildSource(path, mod_prefix + base, None)
-                sources.append(src)
-    return sources
+            mod = module_join(base, mod)
 
+        return mod
 
-def crawl_up(fscache: FileSystemMetaCache, arg: str) -> Tuple[str, str]:
-    """Given a .py[i] filename, return (root directory, module).
+    def crawl_up_dir(self, dir: str) -> str:
+        """Given a directory name, return the corresponding module name.
 
-    We crawl up the path until we find a directory without
-    __init__.py[i], or until we run out of path components.
-    """
-    dir, mod = os.path.split(arg)
-    mod = strip_py(mod) or mod
-    while dir and get_init_file(fscache, dir):
-        dir, base = os.path.split(dir)
-        if not base:
-            break
-        # Ensure that base is a valid python module name
-        if not base.isidentifier():
-            raise InvalidSourceList('{} is not a valid Python package name'.format(base))
-        if mod == '__init__' or not mod:
-            mod = base
-        else:
-            mod = base + '.' + mod
+        Use package_cache to cache results."""
```

Comment:
Move the """ to the next line (by itself). There's no need for a subsequent blank line then.